mesh/block.go

```diff
@@ -1,29 +1,47 @@
 package mesh
 
 import (
-	"time"
 	"github.com/google/uuid"
+	"time"
 )
 
-type BLockID uint32
+type BlockID uint32
 
-var layerCounter uint64 = 0
+var layerCounter uint32 = 0
 
 type Block struct {
-	id         BLockID
-	layerNum   uint64
-	blockVotes map[BLockID]bool
+	id         BlockID
+	layerIndex uint32
+	blockVotes map[BlockID]bool
 	timestamp  time.Time
 	coin       bool
 	data       []byte
 	proVotes   uint64
 	conVotes   uint64
 }
 
-func NewBlock(coin bool, data []byte, ts time.Time) *Block{
+func (b Block) Id() uint32 {
+	return uint32(b.id)
```

Comment:
basically it causes lots of type conversions because block id goes into the protobuf as uint32
 ill try and fix it without going into "conversion hell"